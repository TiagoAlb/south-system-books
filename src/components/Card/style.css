.card {
    transition: all ease 0.5s;
    cursor:pointer;
    border-radius: 2px;
    margin-bottom: 20px;
}

.card.noMenu {
    transition: none;
    -webkit-box-shadow: 0px 0px 27px 12px rgba(0,0,0,0.42);
    -moz-box-shadow: 0px 0px 27px 12px rgba(0,0,0,0.42);
    box-shadow: 0px 0px 27px 12px rgba(0,0,0,0.42);
}

.card.noMenu:hover {
    transition: none;
    transform: none;
    -webkit-box-shadow: 0px 0px 27px 12px rgba(0,0,0,0.42);
    -moz-box-shadow: 0px 0px 27px 12px rgba(0,0,0,0.42);
    box-shadow: 0px 0px 27px 12px rgba(0,0,0,0.42);
}

.card:hover {
    transform: scale(1.3);
    transition: all ease 0.7s;
    -webkit-box-shadow: 0px 0px 20px 12px rgba(0,0,0,1);
    -moz-box-shadow: 0px 0px 20px 12px rgba(0,0,0,1);
    box-shadow: 0px 0px 20px 12px rgba(0,0,0,1);
}

.card-image {
    height: 50vh;
    background-size: cover;
    background-position: center;
    border-radius: 2px;
    
    -webkit-box-shadow: inset 0px 0px 80px -27px rgba(0,0,0,1);
    -moz-box-shadow: inset 0px 0px 80px -27px rgba(0,0,0,1);
    box-shadow: inset 0px 0px 80px -27px rgba(0,0,0,1);
}

@media screen and (min-width: 100px) and (max-width: 576px) {
    .card {
        height: 65vh;
    }

    .card-image {
        height: 65vh;
    }
    
    .card:hover {
        transform: scale(1.01);
        transition: all ease 0.7s;
    }
}