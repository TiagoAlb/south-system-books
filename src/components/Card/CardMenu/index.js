import React, { useState } from 'react'
import { connect } from 'react-redux'
import { mapStateToProps, mapDispatchToProps } from '../../../store/functions/book'
import Typography from '@material-ui/core/Typography'
import IconBtn from '../../IconBtn'
import { formatDate } from '../../../utils/date_time_functions'
import { decreaseText } from '../../../utils/web_functions'
import { card_menu } from '../../../lists/options'
import { saveFavorite, saveRecent } from '../../../utils/api_helper'
import './style.css'

const executeScroll = () => window.scrollTo({ top: 0, behavior: 'smooth' })

const CardMenu = ({ item, changeSelectedBook, changeAlert }) => {
    const location = window.location.href
    const path = location.substring(location.lastIndexOf('/') + 1)

    function Options() {
        return (
            <div className='card-menu-content-options'>
                {card_menu.map((prop, key) => {
                    if (prop.name !== 'favorite') {
                        return (
                            <IconBtn
                                key={key}
                                icon={prop.icon}
                                title={prop.title}
                                click={() => (changeSelectedBook(item),
                                    saveRecent(item.id, item.volumeInfo.title),
                                    executeScroll())
                                } />
                        )
                    } else if (path !== 'favoritos')
                        return (
                            <IconBtn
                                key={key}
                                icon={prop.icon}
                                title={prop.title}
                                click={() => (
                                    saveFavorite(item.id, item.volumeInfo.title),
                                    changeAlert({ open: true, message: 'Livro adicionado aos favoritos.', severity: 'success' })
                                )} />
                        )
                })}
            </div>
        )
    }

    return (
        <div className='card-menu'>
            <div className='card-menu-content'>
                <div className='card-menu-content-header'>
                    <Typography variant='subtitle2'>
                        {decreaseText(item.volumeInfo.title, 50)}
                    </Typography>
                    <Typography variant='caption' gutterBottom>
                        {item.volumeInfo.authors && item.volumeInfo.authors.length > 0 ?
                            item.volumeInfo.authors.map((prop, key) => (
                                key > 0 && key < 4 ? ', ' + prop : key < 4 ? prop : ''
                            )) : 'Autor nÃ£o informado'}
                        {' - ' + formatDate(item.volumeInfo.publishedDate, 'DD/MM/YYYY')}
                    </Typography>
                </div>
                <Options />
            </div>
        </div>
    )
}

export default connect(mapStateToProps, mapDispatchToProps)(CardMenu)