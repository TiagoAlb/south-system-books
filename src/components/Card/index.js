import React, { Fragment } from 'react'
import CardMenu from './CardMenu'
import './style.css'

export default ({ item, noMenu }) => {
    return (
        <div className={`card${noMenu ? ' noMenu' : ''}`}>
            { item ?
                <Fragment>
                    <div className='card-image'
                        style={{
                            backgroundImage: `url(${item.volumeInfo.imageLinks.thumbnail})`
                        }}
                    />
                    {!noMenu ?
                        <CardMenu item={item} />
                        : ''}
                </Fragment>
                : <div className='card-image'
                    style={{
                        backgroundImage: `url(https://i2.wp.com/cenfewc.com.br/wp-content/uploads/2018/03/imagem-nao-disponivel.jpg?fit=600%2C600&ssl=1)`
                    }}
                />
            }
        </div>
    )
}