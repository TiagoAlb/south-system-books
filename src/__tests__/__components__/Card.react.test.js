import React from 'react'
import Card from '../../components/Card'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
    const tree = renderer.create(
        <Card noMenu />).toJSON()
    expect(tree).toMatchSnapshot()
})