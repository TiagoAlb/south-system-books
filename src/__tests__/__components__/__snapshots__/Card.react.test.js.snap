// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card noMenu"
>
  <div
    className="card-image"
    style={
      Object {
        "backgroundImage": "url(https://i2.wp.com/cenfewc.com.br/wp-content/uploads/2018/03/imagem-nao-disponivel.jpg?fit=600%2C600&ssl=1)",
      }
    }
  />
</div>
`;
